// devextreme-react/drawer.js
/*!
 * devextreme-react
 * Version: 18.2.13
 * Build date: Wed May 27 2020
 *
 * Copyright (c) 2012 - 2020 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-react
 */

"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var drawer_1 = require("devextreme/ui/drawer");
var prop_types_1 = require("prop-types");
var component_1 = require("./core/component");
var Drawer = /** @class */ (function (_super) {
    __extends(Drawer, _super);
    function Drawer() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this._WidgetClass = drawer_1.default;
        _this._defaults = {
            defaultOpened: "opened"
        };
        _this._templateProps = [{
                tmplOption: "template",
                render: "render",
                component: "component",
                keyFn: "keyFn"
            }];
        return _this;
    }
    Object.defineProperty(Drawer.prototype, "instance", {
        get: function () {
            return this._instance;
        },
        enumerable: true,
        configurable: true
    });
    return Drawer;
}(component_1.Component));
exports.Drawer = Drawer;
Drawer.propTypes = {
    activeStateEnabled: prop_types_1.PropTypes.bool,
    animationDuration: prop_types_1.PropTypes.number,
    animationEnabled: prop_types_1.PropTypes.bool,
    closeOnOutsideClick: prop_types_1.PropTypes.oneOfType([
        prop_types_1.PropTypes.bool,
        prop_types_1.PropTypes.func
    ]),
    disabled: prop_types_1.PropTypes.bool,
    elementAttr: prop_types_1.PropTypes.object,
    height: prop_types_1.PropTypes.oneOfType([
        prop_types_1.PropTypes.func,
        prop_types_1.PropTypes.number,
        prop_types_1.PropTypes.string
    ]),
    hint: prop_types_1.PropTypes.string,
    hoverStateEnabled: prop_types_1.PropTypes.bool,
    maxSize: prop_types_1.PropTypes.number,
    minSize: prop_types_1.PropTypes.number,
    onDisposing: prop_types_1.PropTypes.func,
    onInitialized: prop_types_1.PropTypes.func,
    onOptionChanged: prop_types_1.PropTypes.func,
    opened: prop_types_1.PropTypes.bool,
    openedStateMode: prop_types_1.PropTypes.oneOf([
        "overlap",
        "shrink",
        "push"
    ]),
    position: prop_types_1.PropTypes.oneOf([
        "left",
        "right",
        "top",
        "bottom",
        "before",
        "after"
    ]),
    revealMode: prop_types_1.PropTypes.oneOf([
        "slide",
        "expand"
    ]),
    rtlEnabled: prop_types_1.PropTypes.bool,
    shading: prop_types_1.PropTypes.bool,
    visible: prop_types_1.PropTypes.bool,
    width: prop_types_1.PropTypes.oneOfType([
        prop_types_1.PropTypes.func,
        prop_types_1.PropTypes.number,
        prop_types_1.PropTypes.string
    ])
};
exports.default = Drawer;
