// devextreme-react/file-uploader.js
/*!
 * devextreme-react
 * Version: 18.2.13
 * Build date: Wed May 27 2020
 *
 * Copyright (c) 2012 - 2020 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-react
 */

"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var file_uploader_1 = require("devextreme/ui/file_uploader");
var prop_types_1 = require("prop-types");
var component_1 = require("./core/component");
var FileUploader = /** @class */ (function (_super) {
    __extends(FileUploader, _super);
    function FileUploader() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this._WidgetClass = file_uploader_1.default;
        _this._defaults = {
            defaultValue: "value"
        };
        return _this;
    }
    Object.defineProperty(FileUploader.prototype, "instance", {
        get: function () {
            return this._instance;
        },
        enumerable: true,
        configurable: true
    });
    return FileUploader;
}(component_1.Component));
exports.FileUploader = FileUploader;
FileUploader.propTypes = {
    accept: prop_types_1.PropTypes.string,
    accessKey: prop_types_1.PropTypes.string,
    activeStateEnabled: prop_types_1.PropTypes.bool,
    allowCanceling: prop_types_1.PropTypes.bool,
    allowedFileExtensions: prop_types_1.PropTypes.array,
    chunkSize: prop_types_1.PropTypes.number,
    disabled: prop_types_1.PropTypes.bool,
    elementAttr: prop_types_1.PropTypes.object,
    focusStateEnabled: prop_types_1.PropTypes.bool,
    height: prop_types_1.PropTypes.oneOfType([
        prop_types_1.PropTypes.func,
        prop_types_1.PropTypes.number,
        prop_types_1.PropTypes.string
    ]),
    hint: prop_types_1.PropTypes.string,
    hoverStateEnabled: prop_types_1.PropTypes.bool,
    invalidFileExtensionMessage: prop_types_1.PropTypes.string,
    invalidMaxFileSizeMessage: prop_types_1.PropTypes.string,
    invalidMinFileSizeMessage: prop_types_1.PropTypes.string,
    isValid: prop_types_1.PropTypes.bool,
    labelText: prop_types_1.PropTypes.string,
    maxFileSize: prop_types_1.PropTypes.number,
    minFileSize: prop_types_1.PropTypes.number,
    multiple: prop_types_1.PropTypes.bool,
    name: prop_types_1.PropTypes.string,
    onContentReady: prop_types_1.PropTypes.func,
    onDisposing: prop_types_1.PropTypes.func,
    onInitialized: prop_types_1.PropTypes.func,
    onOptionChanged: prop_types_1.PropTypes.func,
    onProgress: prop_types_1.PropTypes.func,
    onUploadAborted: prop_types_1.PropTypes.func,
    onUploaded: prop_types_1.PropTypes.func,
    onUploadError: prop_types_1.PropTypes.func,
    onUploadStarted: prop_types_1.PropTypes.func,
    onValueChanged: prop_types_1.PropTypes.func,
    progress: prop_types_1.PropTypes.number,
    readOnly: prop_types_1.PropTypes.bool,
    readyToUploadMessage: prop_types_1.PropTypes.string,
    rtlEnabled: prop_types_1.PropTypes.bool,
    selectButtonText: prop_types_1.PropTypes.string,
    showFileList: prop_types_1.PropTypes.bool,
    tabIndex: prop_types_1.PropTypes.number,
    uploadButtonText: prop_types_1.PropTypes.string,
    uploadedMessage: prop_types_1.PropTypes.string,
    uploadFailedMessage: prop_types_1.PropTypes.string,
    uploadHeaders: prop_types_1.PropTypes.object,
    uploadMethod: prop_types_1.PropTypes.oneOf([
        "POST",
        "PUT"
    ]),
    uploadMode: prop_types_1.PropTypes.oneOf([
        "instantly",
        "useButtons",
        "useForm"
    ]),
    uploadUrl: prop_types_1.PropTypes.string,
    validationError: prop_types_1.PropTypes.object,
    value: prop_types_1.PropTypes.array,
    visible: prop_types_1.PropTypes.bool,
    width: prop_types_1.PropTypes.oneOfType([
        prop_types_1.PropTypes.func,
        prop_types_1.PropTypes.number,
        prop_types_1.PropTypes.string
    ])
};
exports.default = FileUploader;
