// devextreme-react/load-indicator.js
/*!
 * devextreme-react
 * Version: 18.2.13
 * Build date: Wed May 27 2020
 *
 * Copyright (c) 2012 - 2020 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-react
 */

"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var load_indicator_1 = require("devextreme/ui/load_indicator");
var prop_types_1 = require("prop-types");
var component_1 = require("./core/component");
var LoadIndicator = /** @class */ (function (_super) {
    __extends(LoadIndicator, _super);
    function LoadIndicator() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this._WidgetClass = load_indicator_1.default;
        return _this;
    }
    Object.defineProperty(LoadIndicator.prototype, "instance", {
        get: function () {
            return this._instance;
        },
        enumerable: true,
        configurable: true
    });
    return LoadIndicator;
}(component_1.Component));
exports.LoadIndicator = LoadIndicator;
LoadIndicator.propTypes = {
    elementAttr: prop_types_1.PropTypes.object,
    height: prop_types_1.PropTypes.oneOfType([
        prop_types_1.PropTypes.func,
        prop_types_1.PropTypes.number,
        prop_types_1.PropTypes.string
    ]),
    hint: prop_types_1.PropTypes.string,
    indicatorSrc: prop_types_1.PropTypes.string,
    onContentReady: prop_types_1.PropTypes.func,
    onDisposing: prop_types_1.PropTypes.func,
    onInitialized: prop_types_1.PropTypes.func,
    onOptionChanged: prop_types_1.PropTypes.func,
    rtlEnabled: prop_types_1.PropTypes.bool,
    visible: prop_types_1.PropTypes.bool,
    width: prop_types_1.PropTypes.oneOfType([
        prop_types_1.PropTypes.func,
        prop_types_1.PropTypes.number,
        prop_types_1.PropTypes.string
    ])
};
exports.default = LoadIndicator;
