// devextreme-react/progress-bar.js
/*!
 * devextreme-react
 * Version: 18.2.13
 * Build date: Wed May 27 2020
 *
 * Copyright (c) 2012 - 2020 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-react
 */

"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var progress_bar_1 = require("devextreme/ui/progress_bar");
var prop_types_1 = require("prop-types");
var component_1 = require("./core/component");
var ProgressBar = /** @class */ (function (_super) {
    __extends(ProgressBar, _super);
    function ProgressBar() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this._WidgetClass = progress_bar_1.default;
        _this._defaults = {
            defaultValue: "value"
        };
        return _this;
    }
    Object.defineProperty(ProgressBar.prototype, "instance", {
        get: function () {
            return this._instance;
        },
        enumerable: true,
        configurable: true
    });
    return ProgressBar;
}(component_1.Component));
exports.ProgressBar = ProgressBar;
ProgressBar.propTypes = {
    disabled: prop_types_1.PropTypes.bool,
    elementAttr: prop_types_1.PropTypes.object,
    height: prop_types_1.PropTypes.oneOfType([
        prop_types_1.PropTypes.func,
        prop_types_1.PropTypes.number,
        prop_types_1.PropTypes.string
    ]),
    hint: prop_types_1.PropTypes.string,
    hoverStateEnabled: prop_types_1.PropTypes.bool,
    isValid: prop_types_1.PropTypes.bool,
    max: prop_types_1.PropTypes.number,
    min: prop_types_1.PropTypes.number,
    onComplete: prop_types_1.PropTypes.func,
    onContentReady: prop_types_1.PropTypes.func,
    onDisposing: prop_types_1.PropTypes.func,
    onInitialized: prop_types_1.PropTypes.func,
    onOptionChanged: prop_types_1.PropTypes.func,
    onValueChanged: prop_types_1.PropTypes.func,
    readOnly: prop_types_1.PropTypes.bool,
    rtlEnabled: prop_types_1.PropTypes.bool,
    showStatus: prop_types_1.PropTypes.bool,
    statusFormat: prop_types_1.PropTypes.oneOfType([
        prop_types_1.PropTypes.func,
        prop_types_1.PropTypes.string
    ]),
    validationError: prop_types_1.PropTypes.object,
    validationMessageMode: prop_types_1.PropTypes.oneOf([
        "always",
        "auto"
    ]),
    value: prop_types_1.PropTypes.number,
    visible: prop_types_1.PropTypes.bool,
    width: prop_types_1.PropTypes.oneOfType([
        prop_types_1.PropTypes.func,
        prop_types_1.PropTypes.number,
        prop_types_1.PropTypes.string
    ])
};
exports.default = ProgressBar;
